src/gl_context.rs

```diff
@@ -26,22 +32,25 @@ pub struct GLContext<Native> {
 }
 
 impl<Native> GLContext<Native>
-    where Native: NativeGLContextMethods
+    where Native: NativeGLContextMethods,
 {
-    pub fn create(shared_with: Option<&Native::Handle>) -> Result<GLContext<Native>, &'static str> {
+    pub fn create(shared_with: Option<&Native::Handle>) -> Result<Self, &'static str> {
         Self::create_shared_with_dispatcher(shared_with, None)
     }
 
     pub fn create_shared_with_dispatcher(shared_with: Option<&Native::Handle>,
                                          dispatcher: Option<Box<GLContextDispatcher>>)
-        -> Result<GLContext<Native>, &'static str> {
+        -> Result<Self, &'static str> {
         let native_context = try!(Native::create_shared_with_dispatcher(shared_with, dispatcher));
+        // XXX how to handle GLES instantiation?
+        let gl_ = gl::GlFns::load_with(|s| GLContext::<Native>::get_proc_address(s) as *const _);
```

Comment:
> alternatively, I can make it generic, but that cries to avoid the virtual dispatch then

Well, doing static dispatch is still on the table, just for a different time frame - https://github.com/servo/gleam/issues/107
So maybe it's not the worst idea to be ready for that ;)

Really though, If this newly created context is shared with some existing one, we should use Gl function loader if that existing one is GL, and GLES correspondingly. If it's an entirely new one, this is probably where we want the `gleam::GlType` flag to be passed in for run-time context type selection.